src/Expect.elm

```diff
@@ -57,8 +68,63 @@ module Expect
+
+
+### Absolute Tolerance
+
+Let's say we want to figure out if our estimation of pi is precise enough.
+
+Is `3.14` within `0.01` of `pi`? Yes, because `3.13 < pi < 3.15`.
+
+    test "3.14 approximates pi with absolute precision" <| \_ ->
+        3.14 |> Expect.within (Absolute 0.01) pi
+
+
+### Relative Tolerance
+
+What if we also want to know if our circle circumference estimation is close enough?
+
+Let's say our circle has a radius of `r` meters. The formula for circle circumference is `C=2*r*pi`.
+To make the calculations a bit easier ([ahem](https://tauday.com/tau-manifesto)), we'll look at half the circumference; `C/2=r*pi`.
+Is `r * 3.14` within `0.01` of `r * pi`?
+That depends, what does `r` equal? If `r` is `0.01`mm, or `0.00001` meters, we're comparing
+`0.00001 * 3.14 - 0.01 < r * pi < 0.00001 * 3.14 + 0.01` or `-0.0099686 < 0.0000314159 < 0.0100314`.
+That's a huge tolerance! A circumference that is _a thousand times longer_ than we expected would pass that test!
+
+On the other hand, if `r` is very large, we're going to need many more digits of pi.
+For an absolute tolerance of `0.01` and a pi estimation of `3.14`, this expectation only passes if `r < 2*pi`.
+
+If we use a relative tolerance of `0.01` instead, the circle area comparison becomes much better. Is `r * 3.14` within
+`1%` of `r * pi`? Yes! In fact, three digits of pi approximation is always good enough for a 0.1% relative tolerance,
+as long as `r` isn't [too close to zero](https://en.wikipedia.org/wiki/Denormal_number).
+
+    fuzz
+        (floatRange 0.000001 100000)
+        "Circle half-circumference with relative tolerance"
+        (\r -> r * 3.14 |> Expect.within (Relative 0.001) (r * pi))
+
+
+### Trouble with Numbers Near Zero
+
+If you are adding things near zero, you probably want absolute tolerance. If you're comparing values between `-1` and `1`, you should consider using absolute tolerance.
+
+For example: Is `1 + 2 - 3` within `1%` of `0`? Well, if `1`, `2` and `3` have any amount of rounding error, you might not get exactly zero. What is `1%` above and below `0`? Zero. We just lost all tolerance. Even if we hard-code the numbers, we might not get exactly zero; `0.1 + 0.2` rounds to a value just above `0.3`, since computers, counting in binary, cannot write down any of those three numbers using a finite number of digits, just like we cannot write `0.333...` exactly in base 10.
+
+Another example is comparing values that are on either side of zero. `0.0001` is more than `100%` away from `-0.0001`. In fact, `infinity` is closer to `0.0001` than `0.0001` is to `-0.0001`, if you are using a relative tolerance. Twice as close, actually. So even though both `0.0001` and `-0.0001` could be considered very close to zero, they are very far apart relative to each other. The same argument applies for any number of zeroes.
+
+
+### Summary
+
+In general, if you are multiplying, you want relative tolerance, and if you're adding,
```

Comment:
Done. I think I hoped that putting at the end would mean that people read through it at least once, but now I think that would just lead to people not reading even the summary.